server/src/main/java/org/elasticsearch/cluster/routing/allocation/RoutingAllocation.java

```diff
@@ -333,6 +346,33 @@ public void setHasPendingAsyncFetch() {
         this.hasPendingAsyncFetch = true;
     }
 
+    public long unaccountableSearchableSnapshotSize(RoutingNode node) {
+        try {
+            return unaccountableSearchableSnapshotSizes.computeIfAbsent(node, k -> {
+                long totalSize = 0;
+                // Count the STARTED searchable snapshot shards which are unaccounted in the cluster
+                for (ShardRouting shard : node.started()) {
+                    if (clusterInfo == null) {
+                        continue;
+                    }
```

Comment:
Should this not go outside the loop - or even outside the `computeIfAbsent`?

Reply:
Good catch. I've moved it